pennylane/ops/op_math/adjoint_class.py

```diff
@@ -224,3 +232,21 @@ def diagonalizing_gates(self):
 
     def adjoint(self):
         return self.base.queue()
+
+    @property
+    def arithmetic_depth(self) -> int:
+        return 1 + self.base.arithmetic_depth
+
+    def simplify(self, depth=-1):
+        # TODO: Should we add a self.do_queue attribute and instantiate the simplified class
+        # with the same value?
+        if depth == 0:  # TODO: We could add a decorator to do this check for all simplify methods
```

Comment:
Since `if depth==0: return self` is so concise and readable, I think adding a decorator will make the interface more complicated than the code.

It will be harder for developers to remember the syntax for the decorator and track down its definition than to simply remember the code, then it would hurt more than it would help.